Documentation for the new CSV Mappers

labo-flg opened the pull request:
Documenting the solution implemented for https://github.com/OpenCTI-Platform/opencti/issues/2782

RomuDeuxfois commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+# CSV Mappers
+
+CSV Mappers allow to parse CSV files in a given, specific format. The mappers are created and configured by administrators and
+then made available to users who import CSV files, for instance inside a report, and want to extract informations inside these files.
```
```suggestion
then made available to users who import CSV files, for instance inside a report or in global import view, and want to extract informations inside these files.
```

RomuDeuxfois commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+# CSV Mappers
```
We need to add the new capabilites for csv mapper in file capa#list-of-capabilities
The capa is : TAXIIAPI_SETCSVMAPPERS

labo-flg replied:
All good, thanks !

labo-flg commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+## Create a new CSV Mapper
+
+In menu _Data_, select the submenu _Ingestion_, and on the right menu select _CSV Mappers_. You are presented with a list of all mappers set in the platform.
+Note that you can delete or update any mapper from the context menu via the dot icon beside each mapper.
```
```suggestion
Note that you can delete or update any mapper from the context menu via the dot icon beside each and every mapper.
```

Jipegien replied:
```suggestion
Note that you can delete or update any mapper from the context menu via the burger button beside each mapper.
```

Jipegien commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+# CSV Mappers
+
+CSV Mappers allow to parse CSV files in a given, specific format. The mappers are created and configured by administrators and
```
```suggestion
In OpenCTI, CSV Mappers allow to parse CSV files in a STIX 2.1 Objects. The mappers are created and configured by administrators and
```

Jipegien commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+
+CSV Mappers allow to parse CSV files in a given, specific format. The mappers are created and configured by administrators and
+then made available to users who import CSV files, for instance inside a report or in global import view, and want to extract informations inside these files.
+
```
We have to detail here which capability allows a user to use the feature. "Administrator" is too broad

labo-flg replied:
addressed by latest changes
> The mappers are created and configured by users with Manage CSV mappers capability and
then made available to users who import CSV files

Jipegien commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+
+## Principles
+
+The mapper contains __representations__ of Styx entities and relationships, in order for the parser to properly extract them.   
```
```suggestion
The mapper contains __representations__ of STIX 2.1 entities and relationships, in order for the parser to properly extract them.   
```

Jipegien commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,63 @@
+
+In the code block, you will be presented with the raw result of the parsing attempt, in form of a Styx JSON.
+
+You can then check the extracted values if they match the expected entities and relationships.
```
## Use a mapper for importing a CSV file

In Data import section, or Data tab of an entity, when you upload a CSV, you can select a mapper to apply to the file. The file will then be parsed following the rules set in the mapper. 

By default, the imported elements will be added in a new Analyst Workbench where you will be able to check the result of the import.

labo-flg replied:
added

labo-flg commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+
+![Creating a new CSV Mapper](assets/csv-mappers-create.png)
+
+Then, you need to create every representation, one per entity.
```
```suggestion
Then, you need to create every representation, one per entity and relationship type represented in the CSV file.
```

labo-flg commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,72 @@
+
+In the creation or edition form, hit the button __Test__ to open a dialog. Select a sample CSV file and hit the __Test__ button.
+
+In the code block, you will be presented with the raw result of the parsing attempt, in form of a Styx JSON.
```
```suggestion
In the code block, you will be presented with the raw result of the parsing attempt, in form of a STIX 2.1 bundle in JSON format.
```

Archidoit commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,77 @@
+![Creating a new CSV Mapper](assets/csv-mappers-create.png)
+
+Then, you need to create every representation, one per entity and relationship type represented in the CSV file.
+Click the __+__ button to add an empty representation in the list, and click on the chevron to expand the section and configure the representation.
```
Click ON

labo-flg replied:
```suggestion
Click on the __+__ button to add an empty representation in the list, and click on the chevron to expand the section and configure the representation.
```

Archidoit commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,77 @@
+Depending on the entity type, the form will show you the fields that are either required (input outlined in red) or optional.
+For each field, set the corresponding column mapping (the letter-based index of the column in the CSV table, as presented in common spreadsheet tools). 
+
+References to other entities shall be picked from the list of all other representations already defined earlier in the mapper.
```
I would rather say : 'SHOULD be'

Archidoit commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,77 @@
+
+## Use a mapper for importing a CSV file
+
+In Data import section, or Data tab of an entity, when you upload a CSV, you can select a mapper to apply to the file. 
```
in THE Data import section

labo-flg replied:
```suggestion
In the Data import section, or Data tab of an entity, when you upload a CSV, you can select a mapper to apply to the file. 
```

Archidoit commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,77 @@
+## Principles
+
+The mapper contains __representations__ of STIX 2.1 entities and relationships, in order for the parser to properly extract them.   
+One mapper is dedicated to parsing a specific CSV file structure, and thus dedicated mappers shall be created for each 
```
I would rather say: 'SHOULD be created for each'

Archidoit commented:
Very clear 💯

Jipegien commented on docs/usage/import-files.md:
```diff
@@ -3,5 +3,15 @@
     or submit your pull request on the [Github doc repository](https://github.com/OpenCTI-Platform/docs).
+
+
+The `ImportDocument` connector allows to parse several types of files once you have uploaded them, for instance when attaching a file to a report.
+The compatible file types are PDF, plain text, HTML, markdown or CSV.
```
```suggestion
The `ImportDocument` connector allows to parse several types of files once you have uploaded them, for instance when attaching a file to a report.
The compatible file types are PDF, plain text, HTML, markdown or CSV.

Importing file can be done in the `Data` tab of each entity. Upload your file and select the importDocument connector or any more specific connector you may have available. Your file will be associated with the entity your are on. 

It is also possible to import files from the Data Import page, accessible from the top right menu. The file will not be associated with an entity. 

By default, the parsed content of the imported file will be added to a new Analyst Workbench to be reviewed.

Finally, you can also import files in the `Content` tab of [Analyses](exploring-analysis) or [Cases](exploring-cases.md). In this case, the file will be directly added as an attachment and won't use a import connector.

To be able to import file, a user must have the [capability](../administration/users.md) "Upload knowledge files".


```

Jipegien commented on docs/usage/import-files.md:
```diff
@@ -3,5 +3,15 @@
 
     We are doing our best to complete this page. 
-    If you want to participae, dont hesitate to join the [Filigran Community on Slack](https://community.filigran.io) 
+    If you want to participate, dont hesitate to join the [Filigran Community on Slack](https://community.filigran.io) 
```
```suggestion

```

Jipegien commented on docs/usage/import-files.md:
```diff
@@ -3,5 +3,15 @@
     We are doing our best to complete this page. 
-    If you want to participae, dont hesitate to join the [Filigran Community on Slack](https://community.filigran.io) 
+    If you want to participate, dont hesitate to join the [Filigran Community on Slack](https://community.filigran.io) 
     or submit your pull request on the [Github doc repository](https://github.com/OpenCTI-Platform/docs).
```
```suggestion

```

labo-flg commented on docs/usage/assets/Cases-landing-page.png:
removed because duplicates with same file with different casing (lower case first letter)

labo-flg commented on docs/usage/exploring-cases.md:
```diff
@@ -12,7 +12,7 @@ From the `Cases` section, users can access the following tabs:
 - `Feedbacks`: If you use your platform to interact with other teams and provide them CTI Knowledge, some users may want to give you feedback about it. Those feedbacks can easily be considered as an other type of case to solve, as it will often reffer to Knowledge inconsistency or gaps.
 
-![Cases Default page is Incident Response](assets/cases-landing-page.png)
+![Cases Default page is Incident Response](assets/cases-default-landing-page.png)
```
we had twice the same file in the repo, with different casing.
This causes issues in git, the behavior is very unstable with that kind of stuff...

I usually do this : remove all files, use a new one so git detects properly a different file

labo-flg commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,86 @@
+## Principles
+
+The mapper contains __representations__ of STIX 2.1 entities and relationships, in order for the parser to properly extract them.   
+One mapper is dedicated to parsing a specific CSV file structure, and thus dedicated mappers shall be created for each 
```
```suggestion
One mapper is dedicated to parsing a specific CSV file structure, and thus dedicated mappers should be created for each 
```

labo-flg commented on docs/administration/csv-mappers.md:
```diff
@@ -0,0 +1,86 @@
+Depending on the entity type, the form will show you the fields that are either required (input outlined in red) or optional.
+For each field, set the corresponding column mapping (the letter-based index of the column in the CSV table, as presented in common spreadsheet tools). 
+
+References to other entities shall be picked from the list of all other representations already defined earlier in the mapper.
```
```suggestion
References to other entities should be picked from the list of all other representations already defined earlier in the mapper.
```

labo-flg commented:
PR status: *WAITING APPROVAL* 

* I addressed all comments and suggestions
* I removed unrelated changes (the typo fixes everywhere) so we do not defocus this PR (we can do another PR for that)

Archidoit commented on docs/usage/import-files.md:
```diff
@@ -1,7 +1,20 @@
+
+By default, the parsed content of the imported file will be added to a new Analyst Workbench to be reviewed.
+
+Finally, you can also import files in the `Content` tab of [Analyses](exploring-analysis) or [Cases](exploring-cases.md). In this case, the file will be directly added as an attachment and won't use a import connector.
```
```suggestion
Finally, you can also import files in the `Content` tab of [Analyses](exploring-analysis) or [Cases](exploring-cases.md). In this case, the file will be directly added as an attachment and won't use an import connector.
```

Archidoit commented on docs/usage/import-files.md:
```diff
@@ -1,7 +1,20 @@
+
+Finally, you can also import files in the `Content` tab of [Analyses](exploring-analysis) or [Cases](exploring-cases.md). In this case, the file will be directly added as an attachment and won't use a import connector.
+
+To be able to import file, a user must have the [capability](../administration/users.md) "Upload knowledge files".
```
```suggestion
To be able to import a file, a user must have the [capability](../administration/users.md) "Upload knowledge files".
```
